<launch>
  <arg name="mav_name" default="spot"/>
  <arg name="frame_id" default="odom" />
  <arg name="voxel_size" default="0.2" />

  <group ns="$(arg mav_name)">
    <node name="voxblox_sparse_planner" pkg="voxblox_rrt_planner_rt" type="voxblox_sparse_planner_node" output="screen" clear_params="true">
      <!-- feed data generated by the skeleton server -->
      <remap from="voxblox_sparse_planner/esdf_map_in" to="/voxblox_skeleton_server/esdf_map_out" />
      <remap from="voxblox_sparse_planner/tsdf_map_in" to="/voxblox_skeleton_server/tsdf_map_out" />
      <remap from="voxblox_sparse_planner/skeleton_sparse_graph_in" to="/voxblox_skeleton_server/skeleton_sparse_graph_out" />
      <remap from="voxblox_sparse_planner/skeleton_layer_in" to="/voxblox_skeleton_server/skeleton_layer_out" />

      <param name="tsdf_voxel_size" value="$(arg voxel_size)" />
      <param name="tsdf_voxels_per_side" value="16" />

      <param name="visualize" value="true" />
      <param name="robot_radius" value="0.5" />

      <param name="update_mesh_every_n_sec" value="0.0" />
      <param name="num_seconds_to_plan" value="2.0" />
      <param name="do_smoothing" value="true" />
      <param name="frame_id" value="$(arg frame_id)" />
      <param name="world_frame" value="$(arg frame_id)" />
      <param name="trust_approx_solution" value="true" />
      <param name="publish_traversable" value="true" />
      <param name="split_at_collisions" value="true" />
      <param name="optimize_time" value="true" />
      <param name="verbose" value="true" />
      
      <!-- Re-publish ESDF/TSDF maps for the planner -->
      <param name="publish_esdf_map" value="true" />
      <param name="publish_tsdf_map" value="true" />
      <param name="publish_map_every_n_sec" value="1.0" />

      <!-- Re-publish skeleton data -->
      <param name="skeleton_visualize_data" value="true" />
      <param name="skeleton_publish_data" value="true" />
    </node>
  </group>
</launch>
